### PR TITLE
Add unwrap check to lint

### PR DESCRIPTION
**Pull request by zsluedem**

### CODE REVIEW

File: src/types/user_operation.rs [hunk 1]
```diff
@@ -45,20 +52,17 @@ impl From<UserOperation> for entry_point_api::UserOperation {
...
-        );
-        encoded.push_str(&packed[..packed.len() - 64]);
-        encoded.parse::<Bytes>().unwrap()
+        let mut packed: Vec<u8> = UserOperation {
+            signature: Bytes::default(),
+            ..self.clone()
+        }
+        .encode();
+        packed.truncate(packed.len() - 32);
+        Bytes::from(packed)
```
Comment (zsluedem) [inline src/types/user_operation.rs:65]: We don't have to convert from bytes to str and then str to bytes which is unnecessary and slower. We could just operate on vec bytes.

File: src/types/user_operation.rs [hunk 2]
```diff
@@ -5,25 +5,32 @@ use ethers::types::{Address, Bytes, TransactionReceipt, H256, U256};
...
-    pub call_gas_limit: U256,
-    pub verification_gas_limit: U256,
-    pub pre_verification_gas: U256,
-    pub max_fee_per_gas: U256,
-    pub max_priority_fee_per_gas: U256,
-    pub paymaster_and_data: Bytes,
-    pub signature: Bytes,
+// In order to mute unwrap used lint error which is generated by the #derive(EthAbiType)
+// UserOperation is put in the private mod and mute the error.
+#[allow(clippy::unwrap_used)]
```
Comment (zsluedem) [inline src/types/user_operation.rs:13]: EthAbiType would generate unwrap which I could not avoid now.

Comment (zsluedem) [reply src/types/user_operation.rs:13]: https://github.com/gakonst/ethers-rs/issues/2053 created an issue for ethers-rs

Comment (Vid201) [reply src/types/user_operation.rs:13]: @zsluedem Do you want to wait with this PR to see if this fix in `ethers-rs` will be merged in the coming days (we can then just update the lib version)?

File: src/types/user_operation.rs (hunk 1 above)
Comment (Vid201) [reply src/types/user_operation.rs:65]: Looks good

File: src/types/user_operation.rs (hunk 2 above)
Comment (zsluedem) [reply src/types/user_operation.rs:13]: Hm.. I am fine with either way.

Comment (Vid201) [reply src/types/user_operation.rs:13]: Then let's wait.

Comment (zsluedem) [reply src/types/user_operation.rs]: @Vid201 Updated now.

**Review by Vid201 — APPROVED** [review]

**Pull request merged**